File: 342-PowerOfFour/342-PowerOfFour.java
// Last updated: 7/14/2025, 4:35:39 AM
class Solution {
    public boolean isPowerOfFour(int n) {
        if (n <= 0) return false;
        while( n % 4  == 0){
            n = n/4;
        }
        return n == 1;
      
    }
}

File: 342-PowerOfFour/342-PowerOfFour.cpp
// Last updated: 7/16/2025, 9:33:45 PM
class Solution {
public:
    bool isPowerOfFour(int n) {
        if( n <= 0){
            return false;
        }
        while (n % 4 ==0 ){
            n = n/4;
        }
        return n == 1;
    }
};